kong/plugins/moesif/moesif_gov.lua

```diff
@@ -494,13 +500,14 @@ function _M.govern_request(ngx, conf, start_access_phase_time)
         end
     end
 
+    local regex_gov_rules = get_rules(hash_key, RuleType.REGEX, "unidentified", conf)
     -- Check if need to block request based on the regex governance rule
-    if regex_rules ~= nil and type(regex_rules) == "table" and next(regex_rules) ~= nil then
+    if regex_gov_rules ~= nil and type(regex_gov_rules) == "table" and next(regex_gov_rules) ~= nil then
         -- Check if the governance rule regex config matches request config mapping and fetch governance rule id
-        local gr_id = regex_config_helper.fetch_governance_rule_id_on_regex_match(regex_rules, request_config_mapping, conf)
+        local gr_id = regex_config_helper.fetch_governance_rule_id_on_regex_match(regex_gov_rules, request_config_mapping, conf)
         -- Check if need to block request based on governance rule regex config
         if gr_id ~= nil then
-            local sc_regex_req = block_request_based_on_governance_rule_regex_config(hash_key, RuleType.REGEX, gr_id, conf, start_access_phase_time, nil, nil)
+            local sc_regex_req = block_request_based_on_governance_rule_regex_config(regex_gov_rules, hash_key, RuleType.REGEX, gr_id, conf, start_access_phase_time, nil, nil)
```

Comment:
nitpic, `hash_key` should had been first param.

Reply:
Changed